keadm/app/cmd/common/types.go

```diff
@@ -17,24 +17,39 @@ limitations under the License.
 package common
 
 import (
+	"fmt"
 	"time"
+
+	"github.com/kubeedge/kubeedge/common/constants"
 )
 
-//InitOptions has the kubeedge cloud init information filled by CLI
-type InitOptions struct {
+// GlobalOptions is global options for edge and init
+type GlobalOptions struct {
 	KubeEdgeVersion   string
 	KubernetesVersion string
 	DockerVersion     string
-	KubeConfig        string
+}
+
+//InitOptions has the kubeedge cloud init information filled by CLI
+type InitOptions struct {
+	GlobalOptions
+	ControllerYaml *ControllerYaml
+	LogYaml        *LoggingYaml
+	ModulesYaml    *ModulesYaml
 }
 
 //JoinOptions has the kubeedge cloud init information filled by CLI
 type JoinOptions struct {
-	InitOptions
-	CertPath           string
-	EdgeControllerIP   string
+	GlobalOptions
+	EdgeYaml    *EdgeYamlSt
```

Comment:
Can we rename all structures ending with "St" to something more meaningful ?
Such suffixes are not easy to perceive what they stand for .

Reply:
ok